core/src/test/java/com/linecorp/armeria/server/RedirectServiceTest.java

```diff
@@ -0,0 +1,291 @@
+                "/test1i",
+                "/test1j",
+                "/test1k/branch1",
+                "/test1k/branch2",
+
+                "/test2a/qwe/asd/zxc",
+                "/test2b/rty/fgh/vbn"
+        };
+        HttpStatus[] redirectStatuses = new HttpStatus[] {
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.MULTIPLE_CHOICES,
+                HttpStatus.MOVED_PERMANENTLY,
+                HttpStatus.FOUND,
+                HttpStatus.SEE_OTHER,
+                HttpStatus.NOT_MODIFIED,
+                HttpStatus.USE_PROXY,
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.TEMPORARY_REDIRECT,
+
+                HttpStatus.TEMPORARY_REDIRECT,
+                HttpStatus.TEMPORARY_REDIRECT
+        };
+        String[] expectedLocations = new String[] {
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch1",
+                "/new0/branch2",
+
+                "/new1/qwe/new1/asd/new1/zxc",
+                "/new1/rty/new1/fgh/new1/vbn"
+        };
+        String[] expectedResponse = new String[] {
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_1",
+                "SERVICE_BRANCH_2",
+
+                "SERVICE_NAMED_PARAM qwe asd zxc",
+                "SERVICE_NAMED_PARAM rty fgh vbn",
+        };
+
+        HttpClient client = HttpClient.of(serverRule1.uri("/"));
+
+        for (int i = 0; i < testPaths.length; ++i) {
+            AggregatedHttpMessage msg = client.get(testPaths[i]).aggregate().get();
+            assertThat(msg.status()).isEqualTo(redirectStatuses[i]);
+
+            String newLocation = msg.headers().get(HttpHeaderNames.LOCATION);
+            assertThat(newLocation).isEqualTo(expectedLocations[i]);
+
+            msg = client.get(newLocation).aggregate().get();
+            assertThat(msg.status()).isEqualTo(HttpStatus.OK);
+            assertThat(msg.content().toStringUtf8()).isEqualTo(expectedResponse[i]);
+        }
+        serverRule1.stop();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testMisconfiguredIllegalStatus1() throws Exception {
+        new RedirectService(HttpStatus.OK, "/new0/branch1");
```

Comment:
Could you use `assertThatThrownBy` instead of `@Test(expected =...)`?
```
assertThatThrownBy(() -> new RedirectService(HttpStatus.OK, "/new0/branch1"))
.isInstanceOf(IllegalArgumentException.class)
... more checks here ...
```

Reply:
hi @imasahiro PTAL